datasets/chateau.py

```diff
@@ -0,0 +1,39 @@
+from benchopt import BaseDataset, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import deepinv as dinv
+    import torch
```

Comment:
I would avoid using deepinv for the dataset, this makes it harder to run this dataset without torch, which is for most other solvers.
You can create a linear operator A without deepinv.

Also, performing the operation on GPU for such small scale computation (processing the dataset) is not a very good idea as the overhead of transferring the image `x` to GPU and back in memory is probably larger than the gain in time.